src/reynir/verbframe.py

```diff
@@ -248,86 +255,162 @@ def matches_pcl(self, particle: str) -> bool:
+                #    raise ConfigError("Preposition should have exactly one argument")
+                if parg[1] not in ALL_CASES and parg[1] not in SUBCLAUSES:
+                    parg[1] = REFLPRN.get(parg[1], parg[1])
+                    assert parg[1] is not None
+                    spl = parg[1].split("_")
+                    skip = ["gr", "ft", "est", "mst", "et", "kk", "kvk", "hk"]   # TODO Handle these variants
+                    while True:
+                        if spl[-1].strip() in skip:
+                            spl = spl[:-1]
+                        else:
+                            break
+                    if spl[-1] not in ALL_CASES:
+                        raise ConfigError(
+                            "Preposition argument must have a case as its last variant"
+                        )
+                    parg[1] = spl[-1]
+                prepositions.append((parg[0].replace("_", " "), parg[1]))
+                ix += 1
+            return s, prepositions
+
+        def get_direct_object(s: str) -> Tuple[str, str]:
+            # Process direct object argument
+            op = s.split("|")
+            s = op[0]
+            obj: str = ""
+            if len(op) < 1 or len(op) > 2:
+                raise ConfigError("Verb should have zero or one direct object")
+            elif len(op) == 2:
+                pobj = op[1].strip()
+                if pobj not in ALL_CASES and pobj not in SUBCLAUSES:
+                    pobj = pobj.split(" ")[0].strip()
+                    pobj = REFLPRN.get(pobj, pobj)
+                    assert pobj is not None
+                    spl = pobj.split("_")
+                    skip = ["gr", "ft", "est", "mst", "et", "kk", "kvk", "hk"]
+                    while True:
+                        if spl[-1].strip() in skip:
+                            spl = spl[:-1]
+                        else:
+                            break
+                    if spl[-1] not in ALL_CASES:
+                        raise ConfigError(
+                            "Direct object must have a case as its last variant"
+                        )
+                    pobj = spl[-1]
+                obj = pobj
+            return s, obj
+
+        def get_indirect_object(s: str) -> Tuple[str, str]:
+            # Process indirect object argument
+            a = s.split()
+            if len(a) < 1:
+                raise ConfigError("Verb should have zero or one indirect objects")
+            #if len(a) < 1 or len(a) > 3:
+            #    raise ConfigError("Verb should have zero, one or two arguments")
+            verb = a[0]
+            iobj: str = ""
+            if not verb.isalpha():
+                raise ConfigError("Verb '{0}' is not a valid word".format(verb))
+            if len(a) > 1:
+                piobj = " ".join(a[1:]).strip()
+                if piobj not in ALL_CASES and piobj not in SUBCLAUSES:
+                    piobj = piobj.split(" ")[0].strip()
```

Comment:
Again, factor this into a separate function

Reply:
Done